examples/protyping/auto-segmentation/dicom_process.py

```diff
@@ -0,0 +1,383 @@
+# -*- coding: utf-8 -*-
+"""
+
+Document:
+This script aims to take dicom files and prepare them for:
+a) Training / Test data for the model (includes label processing)
+b) Prediction input for the model (model returns the labels)
+
+Example:
+
+Attributes:
+
+Todo:
+"""
+
+from __future__ import print_function
+
+from glob import glob
+
+import numpy as np
+
+import skimage.draw
+import skimage.transform
+
+import pydicom
+
+
+def list_files(data_path, ext=None):
+    """Returns a sorted list of all files names in data_path with (optional) ext.
+
+    Parameters
+    ----------
+    data_path : str
+        Path to list files
+    ext : str
+        Option to list only files with extension .ext
+
+    Returns
+    -------
+    file_names : list
+        Sorted list of file names
+    """
+
+    if ext is None:
+        file_names = glob(data_path + "/*")
+    else:
+        file_names = glob(data_path + "/*" + ext)
+    file_names.sort()
+    return file_names
+
+
+def read_dicom_files(file_names):
+    """
+    Returns a list of DICOM files read in from each file_name
+    in list file_names.
+    """
+    return [pydicom.dcmread(file_name, force=True) for file_name in file_names]
+
+
+def filter_dicom_files(dicom_files):
+    """
+    Filters a list of DICOM files into 4 lists depending on file attributes:
+
+    dicom_series: DICOM imaging series (CT DICOM files).
+
+    dicom_structures: DICOM structure set file (RS DICOM file).
+
+    dicom_plan: DICOM treatment plan file (RP DICOM file).
+
+    dicom_dose: DICOM dose grid file (RD DICOM file).
+    """
+    dicom_series = []
+    dicom_structures = []
+    dicom_plan = []
+    dicom_dose = []
+
+    for file in dicom_files:
+        if hasattr(file, "ImageType"):
+            dicom_series.append(file)
+        elif hasattr(file, "StructureSetName"):
+            dicom_structures.append(file)
+        elif hasattr(file, "BeamSequence"):
+            dicom_plan.append(file)
+        else:
+            # TODO Add condition - will it always be DICOM dose file?
+            dicom_dose.append(file)
```

Comment:
Better pattern here is to explicitly test for DICOM dose, and then have a final else that raises a `ValueError` saying something like "Unsupported DICOM type found."

Reply:
elif hasattr(file, "BeamSequence"):
            dicom_plan.append(file)
        elif hasattr(file, "DoseType"):
            dicom_dose.append(file)    
        else:
            raise ValueError('Unsupported DICOM type found', dicom_files)
    return dicom_series, dicom_structures, dicom_plan, dicom_dose`
`